src/server.cpp

```diff
@@ -670,20 +670,21 @@ void Server::AsyncRunStep(bool initial_step)
 		ScopeProfiler sp(g_profiler, "Server: checking added and deleted objs");
 
 		// Radius inside which objects are active
-		s16 radius = g_settings->getS16("active_object_send_range_blocks");
-		s16 player_radius = g_settings->getS16("player_transfer_distance");
-
-		if (player_radius == 0 && g_settings->exists("unlimited_player_transfer_distance") &&
-				!g_settings->getBool("unlimited_player_transfer_distance"))
+		static const s16 radius =
+			g_settings->getS16("active_object_send_range_blocks") * MAP_BLOCKSIZE;
+
+		// Radius inside which players are active
+		static const bool is_transfer_limited =
+			g_settings->exists("unlimited_player_transfer_distance") &&
+			!g_settings->getBool("unlimited_player_transfer_distance");
+		s16 player_radius =
+			g_settings->getS16("player_transfer_distance") * MAP_BLOCKSIZE;
```

Comment:
Unlimited_player_transfer_distance is a depreciated setting. It's replaced
be player_transfer_distance.
On 13 Sep 2015 03:05, "ShadowNinja" <notifications@github.com> wrote:

> In src/server.cpp
> <https://github.com/minetest/minetest/pull/3140#discussion_r39343638>:
>
> > @@ -670,20 +670,21 @@ void Server::AsyncRunStep(bool initial_step)
> >  		ScopeProfiler sp(g_profiler, "Server: checking added and deleted objs");
> >
> >  		// Radius inside which objects are active
> > -		s16 radius = g_settings->getS16("active_object_send_range_blocks");
> > -		s16 player_radius = g_settings->getS16("player_transfer_distance");
> > -
> > -		if (player_radius == 0 && g_settings->exists("unlimited_player_transfer_distance") &&
> > -				!g_settings->getBool("unlimited_player_transfer_distance"))
> > +		static const s16 radius =
> > +			g_settings->getS16("active_object_send_range_blocks") * MAP_BLOCKSIZE;
> > +
> > +		// Radius inside which players are active
> > +		static const bool is_transfer_limited =
> > +			g_settings->exists("unlimited_player_transfer_distance") &&
> > +			!g_settings->getBool("unlimited_player_transfer_distance");
> > +		s16 player_radius =
> > +			g_settings->getS16("player_transfer_distance") * MAP_BLOCKSIZE;
>
> I think that if you're caching unlimited_player_transfer_distance you
> have to cache the distance too.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/minetest/minetest/pull/3140/files#r39343638>.
>